src/actions/publish.js

```diff
@@ -67,6 +68,37 @@ export function uploadFailed(error: mixed): Action {
   return { type: 'UPLOAD_FAILED', error };
 }
 
+/**
+ * This function returns a profile token from a JWT token, if the passed string
+ * looks like a JWT token. Otherwise it just returns the passed string because
+ * this would be the hash directly, as returned by a previous version of the
+ * server.
+ * In the future when the server will be migrated we'll be able to remove this
+ * fallback.
+ */
+function extractProfileTokenFromJwt(hashOrToken: string): string {
+  if (Jwt.isValidJwtToken(hashOrToken)) {
+    // This is a JWT token, let's extract the hash out of it.
+    const jwtPayload = Jwt.extractAndDecodePayload(hashOrToken);
+    if (!jwtPayload) {
+      throw new Error(
+        `The JWT token that's been returned by the server is incorrect.`
+      );
+    }
+
+    const { profileToken } = jwtPayload;
```

Comment:
Are there any guarantees this will be an object that we can destructure? If not it may error, although we'd probably just surface our own error if that happened, so this is probably fine to just assume. Plus, we control the API.

Reply:
In the previous condition I check for this (that it's not null/undefined); then this destructuring operation can't throw.
Then in the next condition I check if it's undefined which can happen if it's not an object or an object without this property (or the empty string obviously).